### ISSUE TITLE
MSAL Sign out error

### ISSUE THREAD

**Issue by ShilpaGavade** [AzureAD/microsoft-authentication-library-for-objc#1038]
I am using MSAL with intune. Initially sign in and sign out flow was working correctly with MSAL but now, after enabling intune policies, PIN view controller is getting presented at the launch of application, from here I am able to sign in successfully but later when I try to sign out, getting error.

MSALCorrelationIDKey = "-------------ID comes here-------------";
MSALErrorDescriptionKey = "parentViewController has no window! Provide a valid controller with view and window.";
MSALInternalErrorCodeKey = "-42000";

**Comment by ShilpaGavade** [comment]
Intune sdk version: 6.0.5
MSAL Library version : 1.1.5

**Comment by oldalton** [comment]
Can you please make sure that the parentViewController that you're passing to MSAL is valid, retained and has a window: https://azuread.github.io/microsoft-authentication-library-for-objc/Classes/MSALWebviewParameters.html#/c:objc(cs)MSALWebviewParameters(py)parentViewController

**Comment by ShilpaGavade** [comment]
Thanks for your comment, I am not retaining view, it is my login view controller and it is getting dismiss after successful MSAL Authentication.
Logout option is inside the application

**Comment by ShilpaGavade** [comment]
Also, this issue is coming on ios 13

**Comment by antrix1989** [comment]
@ShilpaGavade, you need to pass parentViewController to MSAL to solve this issue. You can pass your current view controller.

**Comment by ShilpaGavade** [comment]
@antrix1989 : Yes, currently did the same thing, passing current viewController as parentViewController and it resolved the issue. Thanks a lot.
Also I am having some questions related to MSAL and intune which we are facing now,  sorry for asking in same thread, 
1) MSAL opens Azure login page in safari browser internally, if we want to open it in edge browser then is it possible? How it works?
2) Also, if managed apps need to open through edge browser then should we set something in application side or it needs configuration from azure portal?
3) Using MSAL getting access token but we need SAML token, is it possible using MSAL library?

**Comment by ShilpaGavade** [comment]
Can I get any help regarding point no. 3?

**Comment by mipetriu** [comment]
@ShilpaGavade We don't support Edge in MSAL, only Safari and embedded webview are supported. Unfortunately we also don't support SAML tokens in MSAL. Microsoft recommends to use OAuth2 and OpenID Connect.

**Comment by ShilpaGavade** [comment]
I am facing issue with organisation data removal policy, as mentioned above I have used both MSAL as well as Intune app protection policy in the app, when I try to login with another user using MSAL, it asks for organisation data removal for first user, when I say yes, it removes data (looks like this) but after that I restart app and try to login another user, getting MSAL error, "(Error Domain=MSALErrorDomain Code=-50000 "(null)" UserInfo={MSALErrorDescriptionKey=Multiple accounts found in cache., MSALInternalErrorCodeKey=-42102})'", it means it is not clearing cookies or something like that, should we handle this manually?

**Comment by rakeshiosdev** [comment]
> I am using MSAL with intune. Initially sign in and sign out flow was working correctly with MSAL but now, after enabling intune policies, PIN view controller is getting presented at the launch of application, from here I am able to sign in successfully but later when I try to sign out, getting error.
> 
> MSALCorrelationIDKey = "-------------ID comes here-------------";
> MSALErrorDescriptionKey = "parentViewController has no window! Provide a valid controller with view and window.";
> MSALInternalErrorCodeKey = "-42000";

Please provide a valid controller like: 
webViewParamaters = MSALWebviewParameters(authPresentationViewController: self)
